Python/aot_ceval_jit.c

```diff
@@ -2246,6 +2309,133 @@ void* jit_func(PyCodeObject* co, PyThreadState* tstate) {
                 deferred_vs_pop1_owned(Dst, arg4_idx);
             }
             deferred_vs_apply(Dst);
+
+            char wrote_inline_cache = 0;
+            if (opcode == CALL_METHOD) {
+                CallMethodHint* hint = Dst->call_method_hints;
+                Dst->call_method_hints = hint->next;
+
+                if (hint->attr && hint->meth_found) {
+                    int num_args = oparg + hint->meth_found; // number of arguments to the function, including a potential "self"
+                    int num_vs_args = num_args + 1; // number of python values; one more than the number of arguments since it includes the callable
+
+                    if (hint->attr->ob_type == &PyMethodDescr_Type) {
+                        PyMethodDescrObject* method = (PyMethodDescrObject*)hint->attr;
+                        void* funcptr = method->d_method->ml_meth;
+
+                        if (funcptr && _PyObject_RealIsSubclass((PyObject*)hint->type, (PyObject *)PyDescr_TYPE(method))) {
+                            wrote_inline_cache = 1;
```

Comment:
if you already set `wrote_inline_cache=1` here the later assignments can be removed.